__tests__/app.test.js

```diff
@@ -50,14 +50,87 @@ describe('4: /api/reviews', () => {
                 });
             });
     });
-    test('GET: 200 - produces a reviews array sorted by date in descending order', () => {
+    test('GET: 200 - produces a reviews array sorted by date in descending order (by default when no query given)', () => {
         return request(app)
             .get('/api/reviews')
             .expect(200)
             .then(({body}) => {
                 expect(body.reviews).toBeSortedBy('created_at', {descending: true});
             });
     });
+    describe('11: /api/reviews queries', () => {
+        test('GET: 200 - when given category query returns reviews array only from that category', () => {
+            return request(app)
+                .get('/api/reviews?category=social deduction')
+                .expect(200)
+                .then(({body}) => {
+                    expect(body.reviews.length).toBe(11)
+                    body.reviews.forEach((review) => {
+                        expect(review.category).toBe('social deduction');
+                    });
+                });
+        });
+        test('GET: 200 - if no category is given, defaults to returning all reviews', () => {
+            return request(app)
+                .get('/api/reviews')
+                .expect(200)
+                .then(({body}) => {
+                    expect(body.reviews.length).toBe(13);
+                });
+        });
+        test('GET: 200 - when given sort by query, returns reviews array sorted by this value', () => {
+            return request(app)
+                .get('/api/reviews?sort_by=designer')
+                .expect(200)
+                .then(({body}) => {
+                    expect(body.reviews).toBeSortedBy('designer', {descending: true});
+                });
+        });
+        test('GET: 200 - when given an order, reviews array is ordered by this value', () => {
+            return request(app)
+                .get('/api/reviews?order=ASC')
+                .expect(200)
+                .then(({body}) => {
+                    expect(body.reviews).toBeSortedBy('created_at', {descending: false})
+                });
+        });
+        test('GET: 400 - category given not valid, not equal to any in the categories table', () => {
+            return request(app)
+                .get('/api/reviews?category=banana')
+                .expect(400)
```

Comment:
This should be a 400 as the category doesn't exist.... but could do in the future!